components/channel_push_notification_section.jsx

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {NotificationLevels, NotificationSections} from 'utils/constants.jsx';
+
+import SettingItemMax from 'components/setting_item_max.jsx';
+import SettingItemMin from 'components/setting_item_min.jsx';
+
+export default class ChannelPushNotificationSection extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Expand if true, else collapse the section
+         */
+        expand: PropTypes.bool.isRequired,
+
+        /**
+         * Current member's push notification level
+         */
+        memberNotificationLevel: PropTypes.string.isRequired,
+
+        /**
+         * Current user's global notification level
+         */
+        globalNotificationLevel: PropTypes.string.isRequired,
+
+        /**
+         * onChange handles update of push notification level
+         */
+        onChange: PropTypes.func.isRequired,
+
+        /**
+         * Submit function to save notification level
+         */
+        onSubmit: PropTypes.func.isRequired,
+
+        /**
+         * Update function to to expand or collapse a section
+         */
+        onUpdateSection: PropTypes.func.isRequired,
+
+        /**
+         * Error string from the server
+         */
+        serverError: PropTypes.string
+    }
+
+    handleOnChange = (e) => {
+        this.props.onChange(e.target.value);
+    }
+
+    handleExpandSection = () => {
+        this.props.onUpdateSection(NotificationSections.PUSH);
+    }
+
+    handleCollapseSection = () => {
+        this.props.onUpdateSection();
+    }
+
+    getGlobalNotifyLevelName(level) {
+        let levelName;
+
+        switch (level) {
+        case NotificationLevels.ALL:
+            levelName = (
+                <FormattedMessage
+                    id='channel_notifications.allActivity'
+                    defaultMessage='For all activity'
+                />
+            );
+            break;
+        case NotificationLevels.MENTION:
+            levelName = (
+                <FormattedMessage
+                    id='channel_notifications.onlyMentions'
+                    defaultMessage='Only for mentions'
+                />
+            );
+            break;
+        default:
+            levelName = (
+                <FormattedMessage
+                    id='channel_notifications.never'
+                    defaultMessage='Never'
+                />
+            );
+        }
+
+        return levelName;
+    }
+
+    getDescribe(globalNotifyLevelName) {
```

Comment:
I think this is duplicated code too [here](https://github.com/mattermost/mattermost-webapp/pull/561/files#diff-33ce27fd0efb4acf8686e180212c6a22R96).